packages/ethereum-storage/src/gas-price-definer.ts

```diff
@@ -65,31 +64,52 @@ export default class GasPriceDefiner {
         ]);
       }
     }
-
+    else if (
+      networkName ===
+      EthereumUtils.getEthereumNetworkNameFromId(StorageTypes.EthereumNetwork.XDAI)
+      ) { 
+      const gasPrice: typeof bigNumber = await this.pollProviders(type, networkName);
+      if (gasPrice.length > 0) {
+        return gasPrice
+      } else {
+        this.logger.warn('Not able to parse the gas fees correctly', [
+          'xdai',
+        ]);
+      }
+    }
     return config.getDefaultEthereumGasPrice();
   }
-
   /**
    * Get all gas prices from the APIs
    * If request to the API fails, no value is added to the array
    *
-   * @param type Gas price type (fast, standard or safe low)
-   * @returns Array containing each gas price
+   * @param type Gas price type (fast, standard or safe low).
+   * @param networkName for verifying  whether on ethereum mainnet or xdai , so as to provide reference gasPrice array
+   * @param gasPriceArray returns the gas fees (as an fixed value / or as an array ) based on the network 
+   *  @returns Array containing each gas price
    */
-  public async pollProviders(type: StorageTypes.GasPriceType): Promise<Array<typeof bigNumber>> {
-    const gasPriceArray: Array<typeof bigNumber> = [];
-
-    for (const gasPriceProvider of this.gasPriceProviderList) {
-      try {
-        // Get the gas price from the provider
-        const providerGasPrice = await gasPriceProvider.getGasPrice(type);
-        gasPriceArray.push(providerGasPrice);
-      } catch (err) {
-        // If the function throws, it means the gas price provider is not available or the value sent is not valid
-        this.logger.warn(err, ['ethereum', 'gas']);
+  public async pollProviders(type: StorageTypes.GasPriceType, networkName?: string): Promise<Array<typeof bigNumber> | typeof bigNumber> {
+    if (networkName === EthereumUtils.getEthereumNetworkNameFromId(StorageTypes.EthereumNetwork.MAINNET)) {
+      const gasPriceArray: Array<typeof bigNumber> = [];
+      for (const gasPriceProvider of this.gasPriceProviderList) {
+        try {
+          const providerGasPrice = await gasPriceProvider.getGasPrice(type);
+          gasPriceArray.push(providerGasPrice);
+            } catch (err) {
+          this.logger.warn(err, ['ethereum', 'gas']); 
+        }
       }
+      return gasPriceArray;
     }
+        else if (networkName === EthereumUtils.getEthereumNetworkNameFromId(StorageTypes.EthereumNetwork.XDAI)) {
 
-    return gasPriceArray;
+      const xdaiprovider: StorageTypes.IGasPriceProvider = new XdaiGasPriceProvider();
+      // simply assign the gas fees .
+      const gasPriceArray: typeof bigNumber = xdaiprovider.getGasPrice(type);
+      return gasPriceArray;
+    }
+    else {
+      this.logger.warn('xdai' + 'invalid-input');
```

Comment:
This should be an error, no ?

The message should be self explanatory. 
It misses the tags.

I don't get the `+` here, looks weird.

Reply:
i will rewrite this code again , because it was antipattern , true it will have to be in try .. catch() with an error or missing the fixed indices . will notify that change . thanks